### ISSUE TITLE
'--completion' should take an argument on which shell to eval for

### ISSUE THREAD

**Issue by gretel** [pypa/pipenv#2045]
##### Expected result

`pipenv` should be usable with [`direnv`](https://github.com/direnv/direnv), i.e. like [`minikube`](https://github.com/kubernetes/minikube) does:

```
eval "$(minikube docker-env --shell bash)"
```
the `shell` parameter exactly does what i would expect for `pipenv`.

##### Actual result

internally, `direnv` uses `bash` spawn to do it's work, and will pass the changes to the actual shell.
therefore, using `direnv` in conjunction with `fish` shell will always detect `fish` as shell.
as the syntax is not compatible, this will fail.

##### Steps to replicate

* use `direnv`
* have `eval "$(pipenv --completion)"` in `.envrc`
* issue `direnv allow`

**Comment by gretel** [comment]
@uranusjr are you a robot? what is that supposed to mean?

there is no fix, no workaround - just a lengthy discussion.

i am going to stick with `virtualenv` cause it is supported well. good luck

**Comment by uranusjr** [comment]
Good point, maybe we should build a robot for this. It is quite tiring to close duplicates.

**Comment by gretel** [comment]
@uranusjr i am glad to be helpful. have a great life

**Comment by techalchemy** [comment]
@gretel if you open an issue that is an exact duplicate of an issue which has already been opened, has a full discussion etc, then the expectation is that you can read that one for resolution. This is pretty common for github issue triage. 

As you can see we handle issues in the _thousands_ but are not many people. If you can’t even read the ‘lengthy discussion’ about how to use pipenv with `direnv`, why it’s not built in, and other things that have already been hashed out, why should we spend our (incredibly limited, btw) time giving you a full response?

If you want to re-post the same topic we have already decided on, you need to tell us why your issue is different and demonstrate that you’ve read the original discussion. Right now you’re just demanding a fix. We literally had this _exact_ discussion. Is the problem that we didn’t resolve it to your liking? If so, uh, sorry.
